data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/service/TopicService.java

```diff
@@ -22,9 +26,15 @@ public TopicService(final KafkaProducerConfig kafkaProducerConfig) {
         this.adminClient = AdminClient.create(SinkPropertyConfigurer.getPropertiesForAdminClient(kafkaProducerConfig));
     }
 
-    public void createTopic(final String topicName, final Integer numberOfPartitions, final Short replicationFactor) {
+    public void createTopic(final String topicName, final Integer numberOfPartitions, final Short replicationFactor, final Optional<Long> maxMessageBytes) {
```

Comment:
Passing an `Optional` as an argument is an anti-pattern. Java does nothing to prevent us from passing in `null`. So please change this to `Long maxMessageBytes`.